_source/_posts/2023-10-12-okta-documentation-chatbot.md

```diff
@@ -0,0 +1,155 @@
+
+{% img blog/okta-documentation-chatbot/oktadevforumsdk.jpeg alt:"OktaDev Forum and SDK logo" height:"400" %}{: .center-image } 
+– Image source: <cite>[Okta Developer GitHub](https://raw.githubusercontent.com/oktadev/.github/main/images/okta-dev-header.png)</cite> 
+
+## Evolution of Oktanaut, a Python Chatbot 
+I developed two versions of Oktanaut. The first version was a specific approach, offering greater precision rather than general knowledge. It built this chatbot using LlamaIndex and trained it on the Okta Developer Documentation. While this version generated more accurate responses due to its focused training, it could not handle diverse questions.
+
+The second version, a versatile approach, utilized the capabilities of the [OpenAI GPT-3.5 API](https://chat.openai.com/auth/login); it generated responses based on a broader sample of user inquiries. This version handled various questions, providing a dynamic, self-learning, and interactive experience.
+
+The trade-off between precision and versatility was obvious when comparing the two versions. The LLamaIndex-based Oktanaut version excelled in accuracy, while the GPT-3.5-powered version generated better responses to a broader set of questions. The final chatbot I chose to demo in this blog uses OpenAI API as it has bigger models to run inquiries and is more user-friendly. 
+
+## Meticulous Training on Okta's Developer Documentation for Enhanced Performance
+For both versions, meticulous training was a necessary building block. Doing so allowed for more accurate responses to edge cases and custom-specific questions. I collected training data, including curated sample questions, answers, and information provided by Okta's Developer Support Engineers. I stored this data in Google Drive to run on [Colab](https://colab.google/). Additionally, I transformed the chatbot into a self-learning entity by storing historical interactions, allowing it to learn from its mistakes. This self-improvement mechanism allowed the chatbot to utilize context information from past responses within the session to provide more informed responses.
+
+
+## Personalizing the Conversational Experience with the Python Chatbot AI
+I programmed the chatbot to greet users with its name, functionality, and purpose. Doing so added a personal touch that will help minimize potential confusion users may have when interacting with the chatbot.
+
+I utilized a Python library called [Panel](https://panel.holoviz.org/getting_started/installation.html) on the frontend to achieve a user-friendly interface. Building the interface was not simple as 1)Panel library has several bugs and glitches, and 2)considering a frontend built-in Python is uncommon. Also, another thing to note, when I considered deployment options, I decided to run Oktanaut on Jupyter Notebook to ensure a simple setup process for everyone.
```

Comment:
```suggestion
I utilized a Python library called [Panel](https://panel.holoviz.org/getting_started/installation.html) on the frontend to achieve a user-friendly interface. Building the interface was not as simple as I first expected. The Panel library has several bugs and glitches, and considering a frontend built-in Python is uncommon. When I considered deployment options, I decided to run Oktanaut on Jupyter Notebook to ensure a simple setup process for everyone.
```

Reply:
idk what "considering a frontend built-in Python is uncommon" means but you can leave it